src/api/app/models/project.rb

```diff
@@ -110,6 +111,43 @@ def autocomplete(search)
   validate :valid_name
   validates :kind, inclusion: { in: %w(standard maintenance maintenance_incident maintenance_release) }
 
+  def self.image_templates
+    Project.local_image_templates + remote_image_templates
+  end
+
+  def self.remote_image_templates
+    result = []
+    Project.remote.each do |project|
+      body = load_from_remote(project, '/image_templates.xml')
+      next if body.blank?
+
+      Xmlhash.parse(body).elements('image_template_project').each do |image_template_project|
+        result << image_template_from_xml(image_template_project)
+      end
+    end
+    result
+  end
+
+  def self.load_from_remote(project, path)
+    Rails.cache.fetch("remote_image_templates_#{project.cache_key}", expires_in: 1.hour) do
+      begin
+        ActiveXML.backend.load_external_url("#{project.remoteurl}#{path}")
+      rescue OpenSSL::SSL::SSLError
+        Rails.logger.error "Remote instance #{project.remoteurl} has no valid SSL certificate"
+        next
+      end
+    end
+  end
+
+  def self.image_template_from_xml(image_template_project)
```

Comment:
A short explanation why the objects are just created temporarily  might make sense